OpenKH/egsTools.cs

```diff
@@ -552,4 +703,32 @@ class MDLX{
 		}
 
 	}
+
+    class IMD
+    {
+        //texturecount is always 1 for IMD so we only need 1 Offset, not a list.
+        public int Offset = 0;
+        public bool Invalid = false;
+        public int TextureCount = 0;
+
+        public IMD(byte[] originalAssetData)
+        {
+            using (MemoryStream ms = new MemoryStream(originalAssetData))
+            {
+                int magic = ms.ReadInt32();
+                if (magic != 1145523529) //IMGD
+                {
+                    Invalid = true;
+                    return;
+                }
+
+                TextureCount = 1;
+                ms.ReadInt32(); //always 256
+                int IMDoffset = ms.ReadInt32(); //offset for image data
+                Offset = IMDoffset + 0x20000000;
+            }
+
+        }
+
+    }
```

Comment:
This is good, just needs to be handled into a generic SDasset class so it can convey the important data (offsets, invalid, asset count).